dts/arm/atmel/samd21.dtsi

```diff
@@ -18,20 +18,30 @@
 		};
 	};
 
-	flash0: flash@0 {
-		compatible = "soc-nv-flash";
-		label = "FLASH_0";
-		reg = <0 0x40000>;
-		write-block-size = <64>;
-	};
-
 	sram0: memory@20000000 {
 		device_type = "memory";
 		compatible = "mmio-sram";
 		reg = <0x20000000 0x8000>;
 	};
 
 	soc {
+		nvmctrl: nvmctrl@41004000  {
+			compatible = "atmel,sam0-nvm-ctrl";
+			label = "FLASH_CTRL";
+			reg = <0x40022000 0x22>;
+			interrupts = <5 0>;
+
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			flash0: flash@0 {
```

Comment:
Should the flash block be a child of nvmctrl or a child of soc?  The internal flash doesn't actually depend on the nvmctrl unit.

Reply:
There's no real good answer here, so we've followed this pattern of having the internal flash be a child of the flash controller.

Reply:
SGTM.